tests/test_micropip.py

```diff
@@ -934,3 +934,181 @@ def test_check_compatible(mock_platform, interp, abi, arch, ctx):
     wheel_name = f"{pkg}-{interp}-{abi}-{arch}.whl"
     with ctx:
         WheelInfo.from_url(wheel_name).check_compatible()
+
+
+@run_in_pyodide()
+def test_persistent_mock_pyodide(selenium_standalone_micropip):  #
+    import sys
+    from importlib.metadata import version as importlib_version
+    from io import StringIO
+
+    from micropip._micropip import add_mock_package
+
+    capture_stdio = StringIO()
+
+    sys.stdout = capture_stdio
+    add_mock_package("test_1", "1.0.0", persistent=True)
+    add_mock_package(
+        "test_2",
+        "1.2.0",
+        modules={
+            "t1": "print('hi from t1')",
+            "t2": """
+        def fn():
+            print("Hello from fn")
+        """,
+        },
+        persistent=True,
+    )
+    import test_1
+
+    dir(test_1)
+
+    import t1
+
+    dir(t1)
+    import t2
+
+    dir(t2)
+
+    t2.fn()
+    assert capture_stdio.getvalue().find("Hello from fn") != -1
+    assert capture_stdio.getvalue().find("hi from t1") != -1
+    assert importlib_version("test_1") == "1.0.0"
+
+
+def test_persistent_mock(monkeypatch, capsys):
+    import site
+    from importlib.metadata import version as importlib_version
+
+    from micropip._micropip import add_mock_package
+
+    with TemporaryDirectory() as tmpdirname:
```

Comment:
There is a [`tmp_path`](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html) fixture for this purpose.